AISKU/Tests/perfTests.ts

```diff
@@ -0,0 +1,363 @@
+        },
+      });
+
+      const init = new ApplicationInsights({
+        config: {
+          instrumentationKey: this._instrumentationKey,
+          extensionConfig: {
+            AppInsightsChannelPlugin: {
+              maxBatchInterval: 2000,
+              maxBatchSizeInBytes: 10 * 1024 * 1024, // 10 MB
+            },
+          },
+        },
+      });
+
+      init.core.setPerfMgr(new PerfManager(this._notificationManager));
+      init.loadAppInsights(false, null, this._notificationManager);
+      this._ai = init;
+
+      // Setup Sinon stuff
+      const sender: Sender = this._ai.appInsights.core.getTransmissionControls()[0][0] as Sender;
+      this.errorSpy = this.sandbox.spy(sender, "_onError");
+      this.successSpy = this.sandbox.spy(sender, "_onSuccess");
+      this.loggingSpy = this.sandbox.stub(
+        this._ai.appInsights.core.logger,
+        "throwInternal"
+      );
+    } catch (e) {
+      console.error("Failed to initialize");
+    }
+  }
+
+  public testCleanup() {
+    this.useFakeServer = true;
+    this.useFakeTimers = true;
+    this._sendNotifications = [];
+    this.successSpy.restore();
+  }
+
+  public registerTests() {
+    this.addPerfTests();
+  }
+
+  private addPerfTests(): void {
+    this.testCase({
+      name: "AISKU Perf Tests -> load AppInsights",
+      test: () => {
+        QUnit.assert.ok(this._sendNotifications.length === 1);
+        QUnit.assert.ok(
+          this._sendNotifications[0].name.indexOf("AISKU.loadAppInsights") !==
+            -1,
+          "AppInsights Load time: " + this._sendNotifications[0].exTime
+        );
+        console.log(
+          "Exec time loadAppInsights: " + this._sendNotifications[0].exTime
+        );
+      },
+    });
+
+    this.testCaseAsync({
+      name: "AISKU Perf Tests -> processTelemetry",
+      stepDelay: 100,
+      steps: [
+        () => {
+          for (let i = 0; i < 100; i++) {
+            let exception = null;
+            try {
+              window["a"]["b"]();
+            } catch (e) {
+              exception = e;
+            }
+
+            Assert.ok(exception);
+
+            this._ai.trackException({ exception });
+            this._ai.trackMetric({
+              name: "test",
+              average: Math.round(100 * Math.random()),
+            });
+            this._ai.trackTrace({ message: "test" });
+          }
+        },
+      ]
+        .concat(this.asserts(300))
+        .concat(() => {
+          let totalTrackExtime = 0;
+          let perfEvents = this._sendNotifications;
+          while (perfEvents.length > 0) {
+            const currEvent = perfEvents.shift();
+            if (currEvent.name.indexOf("processTelemetry") !== -1) {
+              totalTrackExtime += currEvent.exTime;
+            }
+            if (currEvent.childEvts && currEvent.childEvts.length > 0) {
+              perfEvents.push.apply(perfEvents, currEvent.childEvts);
+            }
+          }
+          console.log(
```

Comment:
General comment, we should have something that validates that the test actually managed to find the data it's looking for... Otherwise, it's possible that it could report a lower number than expected (because it found nothing)